receiver/awscontainerinsightreceiver/receiver.go

```diff
@@ -180,16 +192,65 @@ func (acir *awsContainerInsightReceiver) initDcgmScraper(ctx context.Context, ho
 		return nil
 	}
 
+	decoConsumer := prometheusscraper.DecorateConsumer{
+		ContainerOrchestrator: ci.EKS,
+		NextConsumer:          acir.nextConsumer,
+		MetricType:            ci.TypeNeuronContainer,
+		K8sDecorator:          decorator,
+		Logger:                acir.settings.Logger,
+	}
+
+	scraperOpts := prometheusscraper.SimplePrometheusScraperOpts{
+		Ctx:               ctx,
+		TelemetrySettings: acir.settings,
+		Consumer:          &decoConsumer,
+		Host:              host,
+		ScraperConfigs:    gpu.GetScraperConfig(hostinfo),
+		HostInfoProvider:  hostinfo,
+		Logger:            acir.settings.Logger,
+	}
+
 	var err error
-	acir.dcgmScraper, err = gpu.NewDcgmScraper(gpu.DcgmScraperOpts{
+	acir.dcgmScraper, err = prometheusscraper.NewSimplePrometheusScraper(scraperOpts)
+	return err
+}
+
+func (acir *awsContainerInsightReceiver) initNeuronScraper(ctx context.Context, host component.Host, hostinfo *hostInfo.Info, decorator *stores.K8sDecorator) error {
+	// if !acir.config.EnableNeuronMetric {
+	// 	return nil
+	// }
+
+	decoConsumer := prometheusscraper.DecorateConsumer{
+		ContainerOrchestrator: ci.EKS,
+		NextConsumer:          acir.nextConsumer,
+		MetricType:            ci.TypeNeuronContainer,
+		K8sDecorator:          decorator,
+		Logger:                acir.settings.Logger,
+	}
+
+	acir.podResourcesStore = stores.NewPodResourcesStore(acir.settings.Logger)
```

Comment:
You can do it now, or I can do it when I implement EFA, but this should be moved higher up in the main `Start()` method since it will be shared.